handlers/chikka/chikka.go

```diff
@@ -1,3 +1,148 @@
 package chikka
 
-/* no logs */
+import (
+	"context"
+	"fmt"
+	"github.com/nyaruka/courier/utils"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+)
+
+var (
+	sendURL      = "https://post.chikka.com/smsapi/request"
+	maxMsgLength = 160
+)
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+func newHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("CK"), "Chikka")}
+}
+
+func init() {
+	courier.RegisterHandler(newHandler())
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	return s.AddHandlerRoute(h, http.MethodPost, "receive", h.ReceiveMessage)
+}
+
+type moMsg struct {
```

Comment:
I think I've finally settled that the right name for these is `form` at least for cases where we are getting posts or gets.

So then you'd have two forms `outgoingForm` and `incomingForm`, naming the instances `form`. Which then all works nicely with `DecodeAndValidateForm` etc..